scripts/hostcfgd

```diff
@@ -1249,6 +1256,96 @@ class PasswHardening(object):
+        ssh_policies.update(self.policies)
+
+        # set new SSH server policies.
+        self.set_policies(ssh_policies)
+
+    def policies_update(self, key, data, modify_conf=True):
+        syslog.syslog(syslog.LOG_DEBUG, "ssh_policies_update - key: {}".format(key))
+        syslog.syslog(syslog.LOG_DEBUG, "ssh_policies_update - data: {}".format(data))
+
+        if data == {}:
+            self.policies = {}
+        else:
+            if 'ports' in data:
+                data['ports'] = data['ports'].split(',')
+            if key == 'POLICIES':
+                self.policies = data
+
+        if modify_conf:
+            self.modify_conf_file()
+
+    # return first line apperience of pattern - else return number of lines in the file
+    def get_line_num_of_pattern(self, pattern, file_path, find_commented=False):
+        syslog.syslog(syslog.LOG_DEBUG, "looking for pattern {} line in file {}".format(pattern, file_path))
+        return_value = 0
+        with open(file_path, 'r') as f:
+            for (i, line) in enumerate(f):
+                if re.match(pattern, line):
+                    syslog.syslog(syslog.LOG_DEBUG, "found pattern {} in line {}".format(pattern, str(i)))
+                    return i + 1
+                if find_commented and re.match('#' + pattern, line):
+                    syslog.syslog(syslog.LOG_DEBUG, "found pattern {} in line {}".format('#' + pattern, str(i)))
+                    return i + 1
+                return_value = i
+        return return_value
+
+    def handle_ports_set(self, values_list):
+        if not len(values_list) > 0:
+            return True
+        key='ports'
+        for port_num in values_list:
+            if isinstance(port_num, int):
+                syslog.syslog(syslog.LOG_ERR, "port num value {} in wrong format".format(port_num))
+                return False
+            if int(port_num) < SSH_MIN_VALUES[key] or SSH_MAX_VALUES[key] < int(port_num):
+                syslog.syslog(syslog.LOG_ERR, "Ssh {} {} out of range".format('port', port_num))
+                return False
+        port_line_num = self.get_line_num_of_pattern("Port", SSH_CONFG, True)
+        modify_single_file_inplace(SSH_CONFG, ['-E', "/^(#)?Port [0-9]+$/d"])
+
+        for port_num in values_list:
+            # add port in original line
+            modify_single_file_inplace(SSH_CONFG, ["{str(port_line_num)} i Port {str(port_num)}"])
+        return True
+
+    def set_policies(self, ssh_policies):
+        # Ssh server flow
+        # The ssh_policies from CONFIG_DB will be set in the ssh config files /etc/ssh/sshd_config
+
+        for key, value in ssh_policies.items():
+            if key == 'ports':
+                if not self.handle_ports_set(value):
+                    syslog.syslog(syslog.LOG_ERR, "Failed to update sshd config files - wrong port configuration")
+                    return
+            elif int(value) < SSH_MIN_VALUES[key] or SSH_MAX_VALUES[key] < int(value):
```

Comment:
Code need improve, so in future add new key to config DB should not break this check.

If this check should break by unknown key, then need improve code to add error log.